third_party/amd/lib/TritonAMDGPUTransforms/ReorderInstructions.cpp

```diff
@@ -123,10 +150,15 @@ class TritonAMDGPUReorderInstructionsPass
       if (localAlloc->getNumOperands() == 1) {
         if (!localAlloc->hasOneUse())
           return;
-        auto loadOp = localAlloc->getOperand(0).getDefiningOp<tt::LoadOp>();
-        if (!loadOp)
+
+        auto definingOp = localAlloc->getOperand(0).getDefiningOp();
+        // Check if localAlloc is in the loop but it's defining op is outside of
```

Comment:
This wording can be quite confusing--it's not `localAlloc`'s defining op; it's the src tensor's defining op. I'd suggest to be clear on that and also change `definingOp` to `srcTensorOp` maybe.